eclair-core/src/test/scala/fr/acinq/eclair/io/PeerConnectionSpec.scala

```diff
@@ -421,5 +421,35 @@ class PeerConnectionSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike wi
       case _ => fail()
     }
   }
+
+  test("incoming rate limiting") { f =>
+    import f._
+    connect(nodeParams, remoteNodeId, switchboard, router, connection, transport, peerConnection, peer, isPersistent = false)
+    val (_, message) = buildMessage(randomKey(), randomKey(), Nil, Left(Recipient(nodeParams.nodeId, None)), Nil)
+    for (_ <- 1 to 11) {
```

Comment:
Why isn't this `1 to 10` if we allow 10 per second?
I guess this is because the rate limiter's behavior isn't entirely predictable, is it?
This can make it hard to test if we want to be really precise, maybe we should keep the tests more vague.
For example, we could do the following:

- override the throttling to 1 per second
- send 5 messages in a loop: verify that at least one went through but not all of them
- sleep 1 second
- send a message and verify it goes through

This provides less strong guarantees but ensures that rate-limiting "roughly works", and I'm not sure we'll be able to get much more from black-box testing if we don't want our tests to randomly fail.

Reply:
Done